api/src/api/extracts_v1/extract_schema.py

```diff
@@ -0,0 +1,74 @@
+from src.api.schemas.extension import Schema, fields, validators
+from src.api.schemas.response_schema import AbstractResponseSchema
+from src.constants.lookup_constants import ExtractType
+from src.pagination.pagination_schema import generate_pagination_schema
+
+
+class ExtractMetadataFilterV1Schema(Schema):
+    extract_type = fields.String(
+        allow_none=True,
+        metadata={
+            "description": "The type of extract to filter by",
+            "example": "opportunity_data_extract",
+        },
+        validate=[validators.OneOf([e.value for e in ExtractType])],
+    )
+    start_date = fields.Date(
+        allow_none=True,
+        metadata={
+            "description": "The start date for filtering extracts",
+            "example": "2023-10-01",
+        },
+    )
+    end_date = fields.Date(
+        allow_none=True,
+        metadata={
+            "description": "The end date for filtering extracts",
+            "example": "2023-10-07",
+        },
+    )
+
+
+class ExtractMetadataRequestSchema(AbstractResponseSchema):
+    filters = fields.Nested(ExtractMetadataFilterV1Schema())
+    pagination = fields.Nested(
+        generate_pagination_schema(
+            "ExtractMetadataPaginationV1Schema",
+            ["created_at"],
+        ),
+        required=True,
+    )
+
+
+class ExtractMetadataResponseSchema(Schema):
+    extract_metadata_id = fields.Integer(
+        metadata={"description": "The ID of the extract metadata", "example": 1}
+    )
+    extract_type = fields.String(
+        metadata={"description": "The type of extract", "example": "opportunity_data_extract"}
+    )
+    file_name = fields.String(
+        metadata={"description": "The name of the file", "example": "data_extract.csv"}
+    )
+    file_path = fields.String(
```

Comment:
To follow a pattern we did on the opportunity attachments, I think we'll want to name this different than the DB - something like `download_path`

This way we can pre-sign the URL and adjust the path without risk of modifying the DB column value.

On that note - I wonder if it is worth defining a base `FileResponseSchema` class that we can re-use in different endpoint. Worth checking if there is enough overlap.

Reply:
@chouinar See latest changes and see if that's what you had in mind. Added a placeholder to make a signed S3 URL, but we may want to sign the URL somewhere in the service layer rather than in the schema object.